engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -66,9 +65,8 @@ internal object NadelHydrationFieldsBuilder {
                 if (objectTypesAreNotReturnedByActorField) {
                     null
                 } else {
-                    childField.toBuilder()
-                        .objectTypeNames(childField.objectTypeNames.filter { it in actorFieldOverallObjectTypeNames })
-                        .build()
+                    childField.objectTypeNames.removeIf { it !in actorFieldOverallObjectTypeNames }
+                    childField
```

Comment:
double checking that we are happy with a mutative op here?

Reply:
Technically it's fine, since we just ran `deepClone`, but I would prefer if we didn't mutate so the codebase is consistent in how it handles it.